helper/resource/teststep_providers_test.go

```diff
@@ -1778,6 +1782,82 @@ func TestTest_TestStep_ProviderFactories_Import_External_WithPersistMatch(t *tes
 	})
 }
 
+func TestTest_TestStep_ProviderFactories_Import_External_WithPersistMatch_WithPersistWorkingDir(t *testing.T) {
+	var result1, result2 string
+
+	t.Setenv(plugintest.EnvTfAccPersistWorkingDir, "1")
+	workingDir := t.TempDir()
+
+	testSteps := []TestStep{
+		{
+			Config:             `resource "random_password" "test" { length = 12 }`,
+			ResourceName:       "random_password.test",
+			ImportState:        true,
+			ImportStateId:      "Z=:cbrJE?Ltg",
+			ImportStatePersist: true,
+			ImportStateCheck: composeImportStateCheck(
+				testExtractResourceAttrInstanceState("none", "result", &result1),
+			),
+		},
+		{
+			Config: `resource "random_password" "test" { length = 12 }`,
+			Check: ComposeTestCheckFunc(
+				testExtractResourceAttr("random_password.test", "result", &result2),
+				testCheckAttributeValuesEqual(&result1, &result2),
+			),
+		},
+	}
+
+	Test(t, TestCase{
+		ExternalProviders: map[string]ExternalProvider{
+			"random": {
+				Source: "registry.terraform.io/hashicorp/random",
+			},
+		},
+		WorkingDir: workingDir,
+		Steps:      testSteps,
+	})
+
+	workingDirPath := filepath.Dir(workingDir)
+
+	for k := range testSteps {
+		dir := workingDirPath + "_" + strconv.Itoa(k+1)
+
+		dirEntries, err := os.ReadDir(dir)
+		if err != nil {
+			t.Errorf("cannot read dir: %s", dir)
+		}
+
+		var workingDirName string
+
+		// Relies upon convention of a directory being created that is prefixed "work".
+		for _, dirEntry := range dirEntries {
+			if strings.HasPrefix(dirEntry.Name(), "work") && dirEntry.IsDir() {
+				workingDirName = filepath.Join(dir, dirEntry.Name())
+				break
+			}
+		}
+
+		configPlanStateFiles := []string{
+			"terraform_plugin_test.tf",
+			"terraform.tfstate",
+			"tfplan",
+		}
+
+		for i, file := range configPlanStateFiles {
+			// Skip verifying plan for first test step as there is no plan file if the
+			// resource does not already exist.
+			if k == 0 && i > 1 {
+				break
+			}
```

Comment:
Nit: These may be a little easier to read if the check was (substituting `k` for `testStepIndex`)

```go
if testStepIndex == 0 && file == "tfplan" {
  continue
}
```